teapots/choreographer-30fps/src/main/cpp/ChoreographerNativeActivity.cpp

```diff
@@ -337,11 +337,19 @@ void Engine::UnloadResources() { renderer_.Unload(); }
 /**
  * Initialize an EGL context for the current display.
  */
-int Engine::InitDisplay() {
+int Engine::InitDisplay(android_app *app) {
   if (!initialized_resources_) {
     gl_context_->Init(app_->window);
     LoadResources();
     initialized_resources_ = true;
+  } else if(app->window != gl_context_->GetANativeWindow()) {
+    assert(gl_context_->GetANativeWindow());
+    UnloadResources();
+    gl_context_->Invalidate();
+    app_ = app;
+    gl_context_->Init(app->window);
```

Comment:
Below (switching line 348 & 349) would look little bit cleaner?

app_ = app;
gl_context_->Invalidate();
gl_context_->Init(app->window);

Reply:
Yeah, putting "app_ = app" after Invalidate() is safer:  when invalidating, whatever the resources allocated before are created with old window, let them released with the old window handle attached to them still; But we must put the "app" into cached "_app" right before any Init(), so in case there is any hidden usage of it, it is automatically handled with new window handle.  I felt it is safe this way: no need to worry too much about hidden usage of _app in other part of the code.

Reply:
LGTM